codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/serde/SerdeExt.kt

```diff
@@ -216,11 +215,31 @@ fun formatInstant(paramName: String, tsFmt: TimestampFormatTrait.Format, forceSt
  * @param paramName The name of the local identifier to convert to an `Instant`
  * @param tsFmt The timestamp format [paramName] is expected to be converted from
  */
-fun parseInstant(paramName: String, tsFmt: TimestampFormatTrait.Format): String = when (tsFmt) {
-    TimestampFormatTrait.Format.EPOCH_SECONDS -> "Instant.fromEpochSeconds($paramName)"
-    TimestampFormatTrait.Format.DATE_TIME -> "Instant.fromIso8601($paramName)"
-    TimestampFormatTrait.Format.HTTP_DATE -> "Instant.fromRfc5322($paramName)"
-    else -> throw CodegenException("unknown timestamp format: $tsFmt")
+fun KotlinWriter.parseInstantExpr(paramName: String, tsFmt: TimestampFormatTrait.Format): String {
+    val fn = when (tsFmt) {
+        TimestampFormatTrait.Format.EPOCH_SECONDS -> "fromEpochSeconds"
+        TimestampFormatTrait.Format.DATE_TIME -> "fromIso8601"
+        TimestampFormatTrait.Format.HTTP_DATE -> "fromRfc5322"
+        else -> throw CodegenException("unknown timestamp format: $tsFmt")
+    }
+    return format("#T.#L(#L)", RuntimeTypes.Core.Instant, fn, paramName)
+}
+
+fun TimestampFormatTrait.Format.toRuntimeEnum(): String = when (this) {
+    TimestampFormatTrait.Format.EPOCH_SECONDS -> "TimestampFormat.EPOCH_SECONDS"
+    TimestampFormatTrait.Format.DATE_TIME -> "TimestampFormat.ISO_8601"
+    TimestampFormatTrait.Format.HTTP_DATE -> "TimestampFormat.RFC_5322"
+    else -> throw CodegenException("unknown timestamp format: $this")
+}
+
+fun TimestampFormatTrait.Format.toRuntimeEnum(writer: KotlinWriter): String {
+    val enum = when (this) {
+        TimestampFormatTrait.Format.EPOCH_SECONDS -> "EPOCH_SECONDS"
+        TimestampFormatTrait.Format.DATE_TIME -> "ISO_8601"
+        TimestampFormatTrait.Format.HTTP_DATE -> "RFC_5322"
+        TimestampFormatTrait.Format.UNKNOWN -> error("unknown timestamp format trait")
+    }
+    return writer.format("#T.#L", RuntimeTypes.Core.TimestampFormat, enum)
 }
```

Comment:
Used in `XmlParserGenerator`. I moved the existing functions here but didn't refactor everything to use this new one since they don't all have access to writer. In general though I'm trying to move away from referencing strings that don't use symbols so ideally things like `toRuntimeEnum` on line 228 would be removed in favor of this new one that actually handles formatting/importing the right symbols for the expression returned.